File: code/java/Exceptions.java
public class Exceptions {
// BEGIN_CODE
    public static double safe_div(int n, int d)
        throws IllegalArgumentException {

        if(d == 0)
            throw new IllegalArgumentException("Divide by zero");

        return n/d;
    }

	public static void main(String[] args) {
        try {
            System.out.println(safe_div(1, 0));
        } catch(IllegalArgumentException err) {
            System.err.println(err.getMessage());
        }
	}
// END_CODE
}

File: code/rust/exceptions.rs
// BEGIN_CODE
fn safe_div(n: f64, d: f64) -> Result<f64, &'static str> {
    if d == 0.0 {
        return Err("Divide by zero");
    } else {
        return Ok(n/d);
    }
}

fn main() {
    match safe_div(1.0, 0.0) {
        Ok(v) => { println!("{}", v); },
        Err(err) => { println!("{}", err); }
    }
}
// END_CODE

